spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java

```diff
@@ -85,4 +87,12 @@ public MappingMongoConverter mappingMongoConverter(MongoMappingContext context,
 		return mappingConverter;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public ReactiveGridFsTemplate reactiveGridFsTemplate(
+			ReactiveMongoDatabaseFactory reactiveMongoDbFactory,
+			MappingMongoConverter mappingMongoConverter) {
+		return new ReactiveGridFsTemplate(reactiveMongoDbFactory, mappingMongoConverter);
```

Comment:
Ideally, we should configure `ReactiveGridFsTemplate ` with `DataBufferFactory` to share potentially resources.

Reply:
`ReactiveGridFsTemplate` constructor has 4 parameters:

- `DataBufferFactory`

- `ReactiveMongoDatabaseFactory`

- `MappingMongoConverter`

- `bucket`

What value of bucket should be in this case?

Reply:
Defaults to `null`, see https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/ReactiveGridFsTemplate.java#L68.